documentation/src/main/asciidoc/ch09.asciidoc

```diff
@@ -326,6 +326,34 @@ For instance, this might be useful if you want to replace the default message of
 constraint with a more explicit one.
 ====
 
+When dealing with distributed architectures, you might need some tolerance for the clock skew in the temporal constraints
+e.g. `@Past` or `@Future`.
+
+You can set a clock skew tolerance by bootstrapping your `ValidatorFactory` as below:
+
+[[example-using-clock-skew-tolerance]]
+.Using clock skew tolerance
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/BootstrappingTest.java[tags=clockSkewTolerance]
+----
+====
+
+Alternatively, you can define it in the XML configuration by setting the `hibernate.validator.clock_skew_tolerance` property
+in your _META-INF/validation.xml_.
+
+The value of this property must be a `long` defining the duration in milliseconds.
+
+[NOTE]
+====
+When implementing your own temporal constraints, you might need to have access to the clock skew tolerance.
+
+It can be obtained by calling the `HibernateConstraintValidatorInitializationContext#getClockSkewTolerance()` method.
+
+Note that to get access to this context at initialization, your validator has to implement the `HibernateConstraintValidator` contract.
```

Comment:
Do we have/need anything for `HibernateConstraintValidator` in the documentation ?

Reply:
Glad you asked. I asked myself the same question when I wrote this part. I'm not totally sure as it's an incubating API so I'm not sure we want to advertise it.

I tried to think about how to present it but everything I can come with is a bit confusing.

So for now, I wouldn't advertise it apart for the clock skew tolerance usage. I added a note to state it's incubating.

What would be interesting would be to check if other validators could benefit from having some additional elements in the initialization context. If we put it in the spec, we should be careful to expose everything that might be needed.

Reply:
That sounds like a good plan :)